plugin/trino-clickhouse/src/test/java/io/trino/plugin/clickhouse/TestClickHouseDistributedQueries.java

```diff
@@ -0,0 +1,220 @@
+                "(col_required BIGINT," + "col_nullable BIGINT," + "col_default BIGINT,"
+                        + "col_nonnull_default BIGINT ," + "col_required2 BIGINT) ENGINE=Log");
+    }
+
+    @Test
+    public void testInsertWithCoercion()
+    {
+        String tableName = "test_insert_with_coercion_" + randomTableSuffix();
+
+        assertUpdate("CREATE TABLE " + tableName + " (" + "tinyint_column TINYINT, " + "integer_column INTEGER, "
+                + "decimal_column DECIMAL(5, 3), " + "real_column REAL, " + "char_column CHAR(3), "
+                + "bounded_varchar_column VARCHAR(3), " + "unbounded_varchar_column VARCHAR, " + "date_column DATE)");
+
+        assertUpdate(
+                "INSERT INTO " + tableName
+                        + " (tinyint_column, integer_column, decimal_column, real_column) VALUES (1e0, 2e0, 3e0, 4e0)",
+                1);
+        assertUpdate("INSERT INTO " + tableName
+                        + " (char_column, bounded_varchar_column, unbounded_varchar_column) VALUES (CAST('aa     ' AS varchar), CAST('aa     ' AS varchar), CAST('aa     ' AS varchar))",
+                1);
+        assertUpdate(
+                "INSERT INTO " + tableName
+                        + " (char_column, bounded_varchar_column, unbounded_varchar_column) VALUES (NULL, NULL, NULL)",
+                1);
+        assertUpdate("INSERT INTO " + tableName
+                        + " (char_column, bounded_varchar_column, unbounded_varchar_column) VALUES (CAST(NULL AS varchar), CAST(NULL AS varchar), CAST(NULL AS varchar))",
+                1);
+        assertUpdate("INSERT INTO " + tableName + " (date_column) VALUES (TIMESTAMP '2019-11-18 22:13:40')", 1);
+
+        assertQuery("SELECT * FROM " + tableName,
+                "VALUES " + "(1, 2, 3, 4, NULL, NULL, NULL, NULL), "
+                        + "(NULL, NULL, NULL, NULL, 'aa     ', 'aa     ', 'aa     ', NULL), "
+                        + "(NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL), "
+                        + "(NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL), "
+                        + "(NULL, NULL, NULL, NULL, NULL, NULL, NULL, DATE '2019-11-18')");
+
+        assertQueryFails("INSERT INTO " + tableName + " (integer_column) VALUES (3e9)",
+                "Out of range for integer: 3.0E9");
+
+        assertUpdate("DROP TABLE " + tableName);
+    }
+
+    @Override
+    protected boolean supportsArrays()
+    {
+        return false;
+    }
+
+    @Override
+    protected boolean supportsCommentOnTable()
+    {
+        return false;
+    }
+
+    @Override
+    public void testCommentColumn()
+    {
+        // currently does not support
+        // TODO
+        assertThatThrownBy(super::testCommentColumn)
+                .hasMessage("This connector does not support setting column comments");
+        throw new SkipException("Unsupported");
+    }
+
+    @Override
+    public void testRenameTable()
```

Comment:
Is this over-ridden because we don't support CTAS in Clickhouse connector? If so, add a TODO with the reason.